coordinator/src/node.rs

```diff
@@ -434,6 +436,88 @@ impl Node {
         )
     }
 
+    pub async fn resize_position(
+        &self,
+        conn: &mut PgConnection,
+        channel_id: ChannelId,
+        trade_params: &TradeParams,
+    ) -> Result<()> {
+        let channel_id_hex = channel_id.to_hex();
+        let peer_id = trade_params.pubkey;
+
+        let position = db::positions::Position::get_position_by_trader(
+            conn,
+            peer_id,
+            vec![PositionState::Open],
+        )?
+        .with_context(|| {
+            format!("Failed to find open position for channel {channel_id_hex} with peer {peer_id}")
+        })?;
+
+        tracing::info!(
+            ?position,
+            ?trade_params,
+            channel_id = %channel_id_hex,
+            peer_id = %peer_id,
+            "Resizing position",
+        );
+
+        // We start the position resizing by proposing a collaborative settlement of the existing
+        // DLC channel _as if nothing had happened_. That is, the `accept_settlement_amount` is
+        // equal to the amount the accept party originally put into the contract.
+        //
+        // TODO: Use subchannel resize protocol to ensure atomicity, simplify the implementation and
+        // improve UX.
+        {
+            // We don't charge a fee for closing because closing the channel is just an
+            // implementation detail. We will charge the fee when opening the new channel.
+            //
+            // TODO: Do we need to include the previous order-matching fee here?
```

Comment:
I think the answer is yes, but I feel like I need some help calculating the correct fee cc @holzeis.